File: frontend/src/interfaces/CheckStatusResponse.ts
// Mirrors the backend Story 2.6 CheckStatusResponse (BR-07). Messages carry the legacy bundle key +
// verbatim resolved text; the frontend renders `text` and never hardcodes SUC/FLD/WRN strings.

import type { MessageInfo } from '@/interfaces/Schedule1Response'

export default interface CheckStatusResponse {
  readonly requirementsMet: boolean
  readonly errors: readonly MessageInfo[]
  readonly warnings: readonly MessageInfo[]
  // SUC-003 success text when requirementsMet; null otherwise.
  readonly message: MessageInfo | null
}

File: backend/src/main/java/ca/bc/gov/nrs/ilcr/schedule1/dto/CheckStatusResponse.java
package ca.bc.gov.nrs.ilcr.schedule1.dto;

import java.util.List;

/**
 * The Schedule 1 Check Status (BR-07) result (Story 2.6, AD-12) — read-only validation, no status
 * transition. {@code requirementsMet} is true iff {@code errors} is empty (warnings do not block).
 * Each {@code errors}/{@code warnings} entry carries the legacy bundle key plus the verbatim resolved
 * text (label-prefixed, AD-8). {@code message} is the SUC-003 success text when requirements are met,
 * else null.
 *
 * @param requirementsMet whether the schedule passes (no errors)
 * @param errors the blocking missing-field / consistency messages (verbatim), in legacy field order
 * @param warnings the non-blocking advisories (e.g. WRN-002 empty-cost row)
 * @param message the "all requirements met" success message when {@code requirementsMet}, else null
 */
public record CheckStatusResponse(
    boolean requirementsMet,
    List<MessageInfo> errors,
    List<MessageInfo> warnings,
    MessageInfo message) {
}
